lib/UtapiClient.js

```diff
@@ -652,20 +652,23 @@ export default class UtapiClient {
     * @param {string} bucket - bucket name
     * @param {number} timestamp - unix epoch timestamp
     * @param {number} objectSize - size of object in bytes
+    * @param {number} prevObjectSize - previous size of object in bytes if this
+    * action overwrote an existing object
     * @param {callback} [cb] - (optional) callback to call
     * @return {undefined}
     */
-    pushMetricPutObject(bucket, timestamp, objectSize, cb) {
+    pushMetricPutObject(bucket, timestamp, objectSize, prevObjectSize, cb) {
         const callback = cb || this._noop;
         if (this.disableClient) {
             return callback();
         }
+        const storageUtilizedDelta = objectSize - prevObjectSize;
         this.log.trace('pushing metric',
             { method: 'UtapiClient.pushMetricPutObject', bucket, timestamp });
         // update counters
         return this.ds.batch([
             ['incrby', genBucketKey(bucket, 'storageUtilizedCounter'),
-                objectSize],
+                storageUtilizedDelta],
             ['incrby', genBucketKey(bucket, 'incomingBytesCounter'),
                 objectSize],
```

Comment:
Regardless of whether the object is being overwritten, there will be bytes transferred over the network and incoming bytes is capturing just that.